### ISSUE TITLE
Unit Tests for Serialization Fail on Linux GPU/CPU CI Runs w Bun v0.2.0

### ISSUE THREAD

**Issue by cryptodeal** [facebookresearch/shumai#76]
As of the release of Bun v0.2.0, the CI test runs for `serialization.test.ts` fail on Linux CPU & Linux GPU CI Test runs.

Output from `test-arrayfire-gpu-linux`:
```sh
serialization.test.ts:
✓ serialization > 1D Tensor
native code error: Failed to read 4 bytes from input stream! Read 0
                  at load (/home/circleci/project/shumai/cpp/flashlight_binding.cc:265)

SegmentationFault at 0x0800000000000000


----- bun meta -----
Bun v0.2.0 (b542921f) Linux x64 #21-Ubuntu SMP Fri Jul 10 06:53:47 UTC 2020
TestCommand: 
Elapsed: 6859ms | User: 10109ms | Sys: 3789ms
RSS: 69.21MB | Peak: 0.59GB | Commit: 69.21MB | Faults: 125
----- bun meta -----

Ask for #help in https://bun.sh/discord or go to https://bun.sh/issues


Exited with code exit status 139
```

Output from `test-arrayfire-cpu-linux`:
```sh
serialization.test.ts:
✓ serialization > 1D Tensor
12 |   })
13 |   it('2D Tensor', () => {
14 |     const a = sm.randn([8, 8])
15 |     a.save('/tmp/a')
16 |     const b = sm.tensor('/tmp/a')
17 |     expect(areSameShape(a, b)).toBe(true)
        ^
error: Expected: true
	Received: false
      at /home/circleci/project/test/serialization.test.ts:17:4
✗ serialization > 2D Tensor
✓ serialization > Scalar
```

Interestingly, the same test has been passing on `test-arrayfire-cpu-macos`.

The exception is being thrown in `flashlight_binding.cc` here:
```cc
void* load(char* cstr) {
  try {
    LOCK_GUARD
    auto filename = std::string(cstr);
    fl::Tensor tensor;
    fl::load(filename, tensor);
    auto* t = new fl::Tensor(tensor);
    g_bytes_used += t->bytes();
    return t;
  } catch (std::exception const& e) {
    HANDLE_EXCEPTION(e);
  }
}
```

Per feedback from @asilvas, https://github.com/facebookresearch/shumai/pull/72#issuecomment-1279863782, he's unable to reproduce locally with latest flashlight+shumai+bun v0.2 on main. 

I'm also unable to produce locally, but running Darwin/arm64, so wouldn't expect to be able to reproduce the error given it's not failing on OSx CI test runs.

**Comment by asilvas** [comment]
If it helps, this is the exact same error if you load from a file that does not exist.

```
native code error: Failed to read 4 bytes from input stream! Read 0
                  at load (/PATH/shumai/shumai/cpp/flashlight_binding.cc:265)

SegmentationFault at 0x0800000000000000
```

**Comment by cryptodeal** [comment]
Weirdly, I'm unable to replicate this failing test for `serialization` locally when running our unit tests in Ubuntu `22.0.4.1 LTS` w CPU backend on Bun `v0.2.1`. All tests are passing locally.

**Comment by jacobkahn** [comment]
Fixed in 1014720c0331d9d575b04a383fd071954fb5e343, thanks @bwasti
